packages/node/src/lib/subscribers/subscribers.ts

```diff
@@ -110,27 +129,53 @@ export class Subscribers extends WithHttp implements ISubscribers {
     );
   }
 
+  /**
+   * deprecated use markInAppMessageAs instead
+   */
   async markMessageSeen(subscriberId: string, messageId: string) {
-    return await this.http.post(
+    return await this.novu.post(
       `/subscribers/${subscriberId}/messages/markAs`,
       { messageId, mark: { seen: true } }
     );
   }
 
+  /**
+   * deprecated use markInAppMessageAs instead
+   */
   async markMessageRead(subscriberId: string, messageId: string) {
-    return await this.http.post(
+    return await this.novu.post(
       `/subscribers/${subscriberId}/messages/markAs`,
       { messageId, mark: { read: true } }
     );
   }
 
+  async markInAppMessageAs(
+    subscriberId: string,
+    messageId: string,
+    mark: IMarkFields
+  ) {
+    if (!mark.read && !mark.seen) {
+      throw new Error('Either seen or read must be set');
+    }
+
+    return await this.novu.post(
+      `/subscribers/${subscriberId}/messages/markAs`,
+      { messageId, mark }
+    );
+  }
+
   async markMessageActionSeen(
     subscriberId: string,
     messageId: string,
-    type: string
+    type: ButtonTypeEnum,
+    data: IMarkMessageActionFields
```

Comment:
Isn't this a breaking change? It is not optional... What is behind this?

Reply:
Yes a breaking change, even though this method implementation was wrong earlier.
User could send any string as type and no status